### DISCUSSION TITLE
DataGrid MultiSelect : Cannot provide a value for property 'BehaviourProvider'

### DISCUSSION THREAD

**Discussion by rgomezia** [Megabit/Blazorise#3678] [Q&A]
Hi Masters, 
I  update  blazorise to 1.0.2 from 9.x.x.x  .
but i have only one problem , data grid multiselect not work  and send me this error : 
**There is no registered service of type 'Blazorise.IBehaviourProvider'.
System.InvalidOperationException: Cannot provide a value for property 'BehaviourProvider' on type 'Blazorise.DataGrid._DataGridRowMultiSelect`**

I declare all services on my program how the manual indicate  why this happend

TY for all.

```
public class Program
{
    public static async Task Main(string[] args)
    {
        var builder = WebAssemblyHostBuilder.CreateDefault(args);   
        
        builder.Services
            .AddBlazorise(options =>
            {
                options.Immediate = true;
            })
            .AddMaterialProviders()
            .AddMaterialIcons();
            

        builder.Services.AddSingleton<SessionUser>();
        builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
        builder.RootComponents.Add<App>("#app");

        var host = builder.Build();
        await host.RunAsync();
    }
}
```

**Comment by stsrki** [comment]
Did you update all your project dependencies to Blazorise? Maybe you missed one with the older version.

**Reply by rgomezia** [reply]
Hi, Yes, all Nuggets from blazorise are 1.0.2  , my last version was 0.9.5.4

<PackageReference Include="Blazorise.Components" Version="1.0.2" />

    <PackageReference Include="Blazorise.DataGrid" Version="1.0.2" />

    <PackageReference Include="Blazorise.Icons.Material" Version="1.0.2" />

    <PackageReference Include="Blazorise.Material" Version="1.0.2" />

    <PackageReference Include="Blazorise.Snackbar" Version="1.0.2" />

    <PackageReference Include="Blazorise.SpinKit" Version="1.0.2" />

    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.1" />

    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="6.0.1" PrivateAssets="all" />
    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
    <PackageReference Include="System.Net.Http.Json" Version="6.0.0" />`


This line exactly  if i delete  , can see the datagrid , but i need that multiselect

    <DataGridMultiSelectColumn Width="30px"></DataGridMultiSelectColumn>

**Reply by rgomezia** [reply]
This is my grid maybe some parameter its not ok.  
but i see that and looks fine, on  0.9.5.4 works

                <DataGrid TItem="OrdenFabricacion"
                          Data="OrdenesPlanificadas"
                          SelectionMode="DataGridSelectionMode.Multiple"
                          @bind-SelectedRows="@selectedOrdenFabricacion"
                          ShowPager="true"
                          Sortable="true"
                          Filterable="true"
                          Editable="true"
                          PageSize="8"
                          Responsive="true"
                          UseValidation="true"
                          Resizable="true">

                    <DataGridColumns>

                        <DataGridMultiSelectColumn Width="30px"></DataGridMultiSelectColumn>

                        <DataGridColumn TItem="OrdenFabricacion" Field="@nameof( OrdenFabricacion.Area )" Caption="Area">
                            <FilterTemplate>
                                <TextEdit Placeholder="Filtrar" Text="@context.SearchValue?.ToString()" TextChanged="@(v=> context.TriggerFilterChange(v))" />
                            </FilterTemplate>
                        </DataGridColumn>
                        <DataGridColumn TItem="OrdenFabricacion" Field="@nameof( OrdenFabricacion.DocEntry )" Caption="Codigo">
                            <FilterTemplate>
                                <TextEdit Placeholder="Filtrar" Text="@context.SearchValue?.ToString()" TextChanged="@(v=> context.TriggerFilterChange(v))" />
                            </FilterTemplate>
                        </DataGridColumn>
                        <DataGridColumn TItem="OrdenFabricacion" Field="@nameof( OrdenFabricacion.DocNum )" Caption="Numero">
                            <FilterTemplate>
                                <TextEdit Placeholder="Filtrar" Text="@context.SearchValue?.ToString()" TextChanged="@(v=> context.TriggerFilterChange(v))" />
                            </FilterTemplate>
                        </DataGridColumn>
                        <DataGridColumn TItem="OrdenFabricacion" Field="@nameof( OrdenFabricacion.ProdName )" Caption="Producto" Width="250px;">
                            <FilterTemplate>
                                <TextEdit Placeholder="Filtrar" Text="@context.SearchValue?.ToString()" TextChanged="@(v=> context.TriggerFilterChange(v))" />
                            </FilterTemplate>
                        </DataGridColumn>
                    </DataGridColumns>

                    <ButtonRowTemplate>
                        <Button Style="margin-left:25px;" Color="Color.Success" Clicked="@DoSomethingWithList"><Icon Name="IconName.PlusSquare" />Liberar</Button>
                        <Button Color="Color.Dark" Clicked="@DoSomethingWithLastSelect"><Icon Name="IconName.PlusSquare" />Ir al Detalle</Button>
                    </ButtonRowTemplate>

                    <NextPageButtonTemplate><Icon Class="pager" Name="IconName.StepForward" TextColor="TextColor.Black50" /></NextPageButtonTemplate>
                    <PreviousPageButtonTemplate><Icon Class="pager" Name="IconName.StepBackward" TextColor="TextColor.Black50" /></PreviousPageButtonTemplate>
                    <LastPageButtonTemplate><Icon Class="pager" Name="IconName.Forward" TextColor="TextColor.Black50" /></LastPageButtonTemplate>
                    <FirstPageButtonTemplate><Icon Class="pager" Name="IconName.Backward" TextColor="TextColor.Black50" /></FirstPageButtonTemplate>

                </DataGrid>

**Reply by rgomezia** [reply]
@stsrki Hi , on download code example ,  and run the proyect ,   go to the  CONFIGURABLE DATAGRID  its the same problem not working, i use Blazorise with Material , but run boostrap proyect or bulma work very good , the problem its with Material Demo

**Reply by David-Moreira** [reply]
@stsrki might ahve missed the injection on Material?

**Reply by stsrki** [reply]
Yeah, you're right. I missed adding it to the material config. It's going on the list for 1.0.3.
